nft-minter/src/lib.rs

```diff
@@ -2,8 +2,37 @@
 
 elrond_wasm::imports!();
 
+pub mod admin_whitelist;
+pub mod common_storage;
+pub mod nft_attributes_builder;
+pub mod nft_marketplace_interactor;
+pub mod nft_module;
+pub mod royalties;
+pub mod unique_id_mapper;
+
+use common_storage::CollectionId;
+
 #[elrond_wasm::contract]
-pub trait NftMinter {
+pub trait NftMinter:
+    common_storage::CommonStorageModule
+    + admin_whitelist::AdminWhitelistModule
+    + nft_module::NftModule
+    + nft_attributes_builder::NftAttributesBuilderModule
+    + royalties::RoyaltiesModule
+    + nft_marketplace_interactor::NftMarketplaceInteractorModule
+{
     #[init]
-    fn init(&self) {}
+    fn init(
+        &self,
+        parent_collection_id: CollectionId<Self::Api>,
```

Comment:
Is the `parent_collection_id` actually the CID? (the name is confusing). If so, we actually need a CID/collection, ie when a user creates a new collection (brand in our case) it will upload a new complete NFT folder with a different CID - with ids going from 1 to num_items all over again

Reply:
Renamed the parent collection to category, and gave each NFT its own collection ID.